features/include/pcl/features/impl/scurv.hpp

```diff
@@ -0,0 +1,864 @@
+      w1 = ( hsum + h1 ) / hsumt3;
+      w2 = ( hsum + h2 ) / hsumt3;
+      dmax = std::max ( fabs ( del1 ), fabs ( del2 ) );
+      dmin = std::min ( fabs ( del1 ), fabs ( del2 ) );
+      drat1 = del1 / dmax;
+      drat2 = del2 / dmax;
+      d[i-1] = dmin / ( w1 * drat1 + w2 * drat2 );
+    }
+  }
+//
+//  Set D(N) via non-centered three point formula, adjusted to be
+//  shape preserving.
+//
+  w1 = -h2 / hsum;
+  w2 = ( h2 + hsum ) / hsum;
+  d[n-1] = w1 * del1 + w2 * del2;
+
+  if ( this->pchst ( d[n-1], del2 ) <= 0.0 )
+  {
+    d[n-1] = 0.0;
+  }
+  else if ( this->pchst ( del1, del2 ) < 0.0 )
+  {
+//
+//  Need do this check only if monotonicity switches.
+//
+    dmax = 3.0 * del2;
+
+    if ( fabs ( dmax ) < fabs ( d[n-1] ) )
+    {
+      d[n-1] = dmax;
+    }
+
+  }
+  return;
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+template<typename PointInT, typename PointNT, typename PointOutT> void
+pcl::SCurVEstimation<PointInT, PointNT, PointOutT>::compute (PointCloudOut &output)
+{
+  if (! Feature<PointInT, PointOutT>::initCompute())
+  {
+    output.width = output.height = 0;
+    output.points.clear ();
+    return;
+  }
+  // Copy the header
+  output.header = input_->header;
+
+  // Resize the output dataset
+  // Important! We should only allocate precisely how many elements we will need, otherwise
+  // we risk at pre-allocating too much memory which could lead to bad_alloc 
+  // (see http://dev.pointclouds.org/issues/657)
+  output.width = output.height = 1;
+  output.is_dense = input_->is_dense;
+  output.points.resize (1);
+
+  // Perform the actual feature computation
+  this->computeFeature (output);
+
+  Feature<PointInT, PointOutT>::deinitCompute ();
+}
+
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointInT, typename PointNT, typename PointOutT> void
+pcl::SCurVEstimation<PointInT, PointNT, PointOutT>::computeFeature (PointCloudOut &output)
+{
+  std::cout.precision(3);
+
+  // need to use this vector for storing the nearest 6 points indices
+  std::vector<int> pointIndices;//(6);
+  // need to use this vector for storing the squared distances between each of those
+  // 6 nearest neighbors
+  std::vector<float> euclideanDistances;//(6);
+
+
+  int min_rad = 0;
+  int max_rad = 300;
+  int nrays = 20;
+  int R = (int) ((max_rad - min_rad) / nrays);
+  std::vector<int> R_bins(21);
+  std::vector<int> bin_rad(21);
+  std::vector<double> bins(10);
+  for (int r = 0; r < 21; r++)
+  {
+    R_bins[r] = r * 15;
+    if (r < 10)
+    {
+      bins[9 - r] = 1 - log10(r + 1);
+    }
+  }
+  
+  pcl::PointCloud<pcl::PointNormal>::Ptr input_cloud(new pcl::PointCloud<pcl::PointNormal>);
+  copyPointCloud(*input_, *input_cloud);
+
+  int cnt = 0;
+  Eigen::MatrixXd k1(input_cloud->height,R_bins.size());
+  Eigen::MatrixXd k2(input_cloud->height,R_bins.size());
```

Comment:
Do your input point clouds need to be organized i.e., have a proper height and width?